### PR TITLE
update Berlin crawler

### PR DESCRIPTION
**Pull request by MartinGer**
Add Berlin crawler to pipeline with correct ID, also crawl schools address

### CODE REVIEW

**Comment by MartinGer** [comment]
Updatet to current master

File: jedeschule/spiders/berlin.py [hunk 1]
```diff
@@ -99,4 +100,24 @@ def parse_table(self, table):
...
+    def fix_data(self, string):
+        if string:
+            string = ' '.join(string.split())
+            string.replace('\n', '')
+            string.replace('\t', '')
+        return string
+
+    def normalize(self, item: Item) -> School:
+        return School(name=item.get('name'),
+                      id='BER-{}'.format(item.get('id')),
```
Comment (cyroxx) [inline jedeschule/spiders/berlin.py]: This should be `BE` (derived from the ISO 3166-2:DE code) instead of `BER`.

**Comment by cyroxx** [comment]
At least the headmaster(/director) parsing is not None-safe:

```
2019-11-30 02:40:41 [scrapy.core.scraper] ERROR: Spider error processing <GET http://www.berlin.de/sen/bildung/schule/berliner-schulen/schulverzeichnis/SchulListe.aspx/../Schulportrait.aspx?IDSchulzweig=%2021140> (referer: http://www.berlin.de/sen/bildung/schule/berliner-schulen/schulverzeichnis/SchulListe.aspx/)
Traceback (most recent call last):
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/home/cyroxx/.virtualenvs/jedeschule/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/media/Daten_Linux/Entwicklung/jedeschule-scraper/jedeschule/spiders/berlin.py", line 35, in parse_detail
    headmaster = response.css('#ContentPlaceHolderMenuListe_lblLeitung::text').extract_first().strip()
AttributeError: 'NoneType' object has no attribute 'strip'
```

**Comment by cyroxx** [comment]
@MartinGer What settings did you use for the scraper? After a short moment, I get HTTP 429 (Too Many Requests) errors. Did you have some settings to slow down the scraper so that it behaves nicely?

File: jedeschule/spiders/berlin.py (hunk 1 above)
Comment (MartinGer) [reply jedeschule/spiders/berlin.py]: fixed to BE

**Comment by MartinGer** [comment]
fixed headmaster

> @MartinGer What settings did you use for the scraper? After a short moment, I get HTTP 429 (Too Many Requests) errors. Did you have some settings to slow down the scraper so that it behaves nicely?

I didn't experience any issues while testing with "scrapy crawl Berlin". I used the default settings and didn't change anything. Can anybody else verify the issue?

**Comment by cyroxx** [comment]
Using 7763ffa, I get the following result: 

scrapy result

In this run, I could successfully scrape 523 schools (NOTE that a school that has multiple school types (Berufsfachschule, Fachoberschule) currently appear multiple times),

But for example the [Freudberg Gemeinschaftsschule](https://www.berlin.de/sen/bildung/schule/berliner-schulen/schulverzeichnis/Schulportrait.aspx?IDSchulzweig=21184) was not included in the scraped dataset. I think this is due to the 429 errors.
We should look into [AutoThrottling](https://docs.scrapy.org/en/latest/topics/autothrottle.html) to play nicely with the Berlin website.

**Comment by cyroxx** [comment]
Superseeded by #81

**Pull request closed without merging**